mlflow/gateway/runner.py

```diff
@@ -14,68 +13,95 @@
 _logger = logging.getLogger(__name__)
 
 
-def _monitor_config(config_path: str):
-    config_path = Path(config_path)
-    prev_config = config_path.read_text()
-    for changes in watch(
-        config_path,
-    ):
-        if not any((path == str(config_path)) for _, path in changes):
+def monitor_config(config_path: str) -> Generator[None, None, None]:
+    with open(config_path) as f:
+        prev_config = f.read()
+
+    for changes in watch(os.path.dirname(config_path)):
```

Comment:
We need to watch the directory instead of the file to detect config recreation.

Reply:
I thought that the `watch` function can accept a single file according to its docs: https://watchfiles.helpmanual.io/api/watch/

Reply:
```
> git diff
diff --git a/mlflow/gateway/runner.py b/mlflow/gateway/runner.py
index fcd960f25..2850ac3a4 100644
--- a/mlflow/gateway/runner.py
+++ b/mlflow/gateway/runner.py
@@ -17,7 +17,7 @@ def monitor_config(config_path: str) -> Generator[None, None, None]:
     with open(config_path) as f:
         prev_config = f.read()
 
-    for changes in watch(os.path.dirname(config_path)):
+    for changes in watch(config_path):
         if not any((path == config_path) for _, path in changes):
             continue
 
> pytest tests/gateway/test_runner.py
=================================================================== test session starts ====================================================================
platform linux -- Python 3.8.16, pytest-7.2.0, pluggy-1.0.0 -- /home/haru/miniconda3/envs/mlflow-dev-env/bin/python
cachedir: .pytest_cache
rootdir: /home/haru/Desktop/repositories/mlflow, configfile: pytest.ini
plugins: cov-4.0.0, anyio-3.6.2, localserver-0.5.0
collected 6 items                                                                                                                                          

tests/gateway/test_runner.py::test_server_update PASSED                                                                                              [ 16%]
tests/gateway/test_runner.py::test_server_update_with_invalid_config PASSED                                                                          [ 33%]
tests/gateway/test_runner.py::test_server_update_config_removed_then_recreated FAILED                                                                [ 50%]
tests/gateway/test_runner.py::test_server_static_endpoints PASSED                                                                                    [ 66%]
tests/gateway/test_runner.py::test_server_dynamic_endpoints PASSED                                                                                   [ 83%]
tests/gateway/test_runner.py::test_request_invalid_route PASSED                                                                                      [100%]
```

Reply:
If we pass a config path, it doesn't detect fine recreation.

Reply:
Seems like their docs are just plain wrong. No concerns for me with your implementation :)